platform/linux-generic/odp_queue_basic.c

```diff
@@ -370,6 +519,19 @@ static odp_queue_t queue_create(const char *name,
 		}
 	}
 
+	/* Event aggregator */
+	if (param->num_aggr) {
+		queue_entry_t *aggr_queue = event_aggr_reserve(_odp_queue_glb);
+
+		if (aggr_queue == NULL) {
+			queue->status = QUEUE_STATUS_FREE;
```

Comment:
This causes a potential data race with the queue->status checking in the allocation loop. But there are already data races in the existing code though, between checking the status and writing to it inside and outside the locked section in this function and maybe elsewhere (scheduler) too.

I am not sure what, if anything, can or needs to be done in the context of this PR. Maybe just add a small comment somewhere (like at the place where queue->status is read for the first time), that things are not strictly correct?. Some later change could then improve things (make the status atomic type or something).

Having a separate lock and freelist for (normal) queue allocation might also be worthwhile, but not really in the topic of this PR.